client/rest-high-level/src/main/java/org/elasticsearch/client/ilm/IndexLifecycleExplainResponse.java

```diff
@@ -0,0 +1,359 @@
+    }
+
+    public String getPolicyName() {
+        return policyName;
+    }
+
+    public long getLifecycleDate() {
+        return lifecycleDate;
+    }
+
+    public String getPhase() {
+        return phase;
+    }
+
+    public long getPhaseTime() {
+        return phaseTime;
+    }
+
+    public String getAction() {
+        return action;
+    }
+
+    public long getActionTime() {
+        return actionTime;
+    }
+
+    public String getStep() {
+        return step;
+    }
+
+    public long getStepTime() {
+        return stepTime;
+    }
+
+    public String getFailedStep() {
+        return failedStep;
+    }
+
+    public BytesReference getStepInfo() {
+        return stepInfo;
+    }
+
+    public PhaseExecutionInfo getPhaseExecutionInfo() {
+        return phaseExecutionInfo;
+    }
+
+    public TimeValue getAge() {
+        if (lifecycleDate == null) {
+            return TimeValue.MINUS_ONE;
+        } else {
+            return TimeValue.timeValueMillis(System.currentTimeMillis() - lifecycleDate);
+        }
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        builder.field(INDEX_FIELD.getPreferredName(), index);
+        builder.field(MANAGED_BY_ILM_FIELD.getPreferredName(), managedByILM);
+        if (managedByILM) {
+            builder.field(POLICY_NAME_FIELD.getPreferredName(), policyName);
+            if (lifecycleDate != null) {
+                builder.timeField(LIFECYCLE_DATE_MILLIS_FIELD.getPreferredName(), LIFECYCLE_DATE_FIELD.getPreferredName(), lifecycleDate);
+                builder.field(AGE_FIELD.getPreferredName(), getAge().toHumanReadableString(2));
+            }
+            if (phase != null) {
+                builder.field(PHASE_FIELD.getPreferredName(), phase);
+            }
+            if (phaseTime != null) {
+                builder.timeField(PHASE_TIME_MILLIS_FIELD.getPreferredName(), PHASE_TIME_FIELD.getPreferredName(), phaseTime);
+            }
+            if (action != null) {
+                builder.field(ACTION_FIELD.getPreferredName(), action);
+            }
+            if (actionTime != null) {
+                builder.timeField(ACTION_TIME_MILLIS_FIELD.getPreferredName(), ACTION_TIME_FIELD.getPreferredName(), actionTime);
+            }
+            if (step != null) {
+                builder.field(STEP_FIELD.getPreferredName(), step);
+            }
+            if (stepTime != null) {
+                builder.timeField(STEP_TIME_MILLIS_FIELD.getPreferredName(), STEP_TIME_FIELD.getPreferredName(), stepTime);
+            }
+            if (Strings.hasLength(failedStep)) {
+                builder.field(FAILED_STEP_FIELD.getPreferredName(), failedStep);
+            }
+            if (stepInfo != null && stepInfo.length() > 0) {
+                builder.rawField(STEP_INFO_FIELD.getPreferredName(), stepInfo.streamInput(), XContentType.JSON);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource returned by the following method call: *streamInput*. Currently, there are execution paths that do not contain closure statements, for example, when XContentBuilder.rawField throws an exception. Ensure the object returned by *streamInput()* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).